src/AdditionalPylonsModule.cpp

```diff
@@ -80,7 +80,13 @@ void AdditionalPylonsModule::onUnitCreate(BWAPI::Unit unit) {
 		player.onUnitCreate(unit);
 	}
 	else if(unit->getPlayer() != BWAPI::Broodwar->neutral()){
-		enemy.onUnitCreate(unit);
+		if (unit->getPlayer()->getRace() != enemy.getRace()) {
+			enemy.onUnitCreate(unit);
```

Comment:
no need to have `enemy.onUnitCreate()` in both conditionals, just take it outside the inner conditional